retrieval/discovery/kubernetes/discovery.go

```diff
@@ -265,7 +265,13 @@ func (kd *Discovery) updateNodesTargetGroup() *config.TargetGroup {
 
 	// Now let's loop through the nodes & add them to the target group with appropriate labels.
 	for nodeName, node := range kd.nodes {
-		address := fmt.Sprintf("%s:%d", node.Status.Addresses[0].Address, kd.Conf.KubeletPort)
+		nodeAddress, err := nodeHostIP(node)
+		if err != nil {
+			log.Errorf("Skipping node %s: %s", node.Name, err)
```

Comment:
Can do if this is the approach taken elsewhere to help users identify why a target you'd expect to be scraped isn't added to targets?